test/test-mc-cmp.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2018-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// `test-mc-cmp.h` is a modified copy of `test-bson-cmp.c` from libbson 1.28.0.
+
+#include "test-mongocrypt.h"
+
+#include "mc-cmp-private.h"
+
+static void test_mc_cmp_equal(_mongocrypt_tester_t *tester) {
+    BSON_UNUSED(tester);
+
+    BSON_ASSERT(mc_cmp_equal_ss(0, 0));
```

Comment:
```suggestion
    ASSERT(mc_cmp_equal_ss(0, 0));
```
Use test assertion instead of contract assertion in tests.